records/generator.go

```diff
@@ -108,9 +115,10 @@ func (rg *RecordGenerator) findMaster(masters ...string) (state.State, error) {
 }
 
 // Loads state.json from mesos master
-func (rg *RecordGenerator) loadFromMaster(ip string, port string) (sj state.State) {
+func (rg *RecordGenerator) loadFromMaster(ip string, port string) (*state.State, error) {
```

Comment:
if you're going to return error's here, why bother logging them in this func? leave that up to the caller (and if the caller doesn't log (but also just passes the error back) then it's the grand-caller's job to log, so on and so forth).